command/server.go

```diff
@@ -129,6 +133,33 @@ func (c *ServerCommand) Run(args []string) int {
 		dev = true
 	}
 
+	// Record the vault binary's location and SHA-256 checksum for use in
+	// builtin plugins.
+	ex, err := os.Executable()
+	if err != nil {
+		c.Ui.Output(fmt.Sprintf(
+			"Error looking up vault binary: %s", err))
+		return 1
+	}
+
+	file, err := os.Open(ex)
+	if err != nil {
+		c.Ui.Output(fmt.Sprintf(
+			"Error loading vault binary: %s", err))
+		return 1
+	}
+	defer file.Close()
```

Comment:
This function never exits until Vault shuts down so you'll have an open file handle the whole time. Probably just close it directly after the io.Copy before error checking.